x/concentrated-liquidity/position.go

```diff
@@ -204,10 +215,16 @@ func (k Keeper) deletePosition(ctx sdk.Context,
 	}
 	store.Delete(key)
 
+	// Remove the position ID to underlying lock ID mapping (if it exists)
+	key = types.KeyPositionIdForLock(positionId)
+	if store.Has(key) {
+		store.Delete(key)
+	}
```

Comment:
I think we should be covering this logic with tests as well. Having such areas untested would make any potential refactors much easier because a small accidental change would be caught by the direct unit test